HetDepot/Settings/SettingService.cs

```diff
@@ -28,21 +35,87 @@ public HashSet<String> GetTourTimes()
         public string GetConsoleText(string name, Dictionary<string, string>? components = null)
         {
             /*
-			Get console text from settings. This may contain {components}, which will be replaced with their content in the "components" dictionary, if provided
-			*/
+            Get console text from settings. This may contain {components}, which will be replaced with their content in the "components" dictionary, if provided
+            */
             try
             {
                 if (components == null)
                     return _settings.ConsoleText[name];
-                return (new Regex(@"\{(\w+)\}")).Replace(_settings.ConsoleText[name], match => components.ContainsKey(match.Groups[1].Captures[0].ToString()) ? components[match.Groups[1].Captures[0].ToString()] : match.Value);
+                return (new Regex(@"\{(\w+)\}")).Replace(_settings.ConsoleText[name],
+                    match => components.ContainsKey(match.Groups[1].Captures[0].ToString())
+                        ? components[match.Groups[1].Captures[0].ToString()]
+                        : match.Value);
             }
             catch (Exception e)
             {
                 _errorLogger.LogError($"{this.GetType()} - Input [{name}] - {e.Message}");
             }
 
             // return string.Empty; //TODO: nadenken of dit eigen erro moet worden
-            return name; // Liever gewoon fallback naar de "name", dan staat er tenminste iets toch? (Idem in de regex hierboven)
+            return
+                name; // Liever gewoon fallback naar de "name", dan staat er tenminste iets toch? (Idem in de regex hierboven)
+        }
+
+        private void _initializeTourStorage()
+        {
+            var settingsDir = GetSettingDir();
+            string path = GetToursPath();
+
+            if (!Directory.Exists(settingsDir))
+            {
+                Directory.CreateDirectory(settingsDir);
+            }
+
+            if (File.Exists(path))
+            {
+                // Do not generate tours
+                return;
+            }
+
+            var createdFile = File.Create(path);
+            createdFile.Close();
+
+            var tourTimes = GetTourTimes();
+
+            var tours = tourTimes.Select((string time) =>
+            {
+                DateTime today = DateTime.Today;
+                int hours = Int32.Parse(time.Split(':')[0]);
+                int minutes = Int32.Parse(time.Split(':')[1]);
+
+                return new Tour(
+                    new DateTime(today.Year, today.Month, today.Day, hours, minutes, 0),
+                    GetMaxTourReservations(),
+                    new List<Visitor>(),
+                    new List<Visitor>());
+            }).ToList();
+
+            StreamWriter writer = new StreamWriter(path);
+            writer.Write(JsonSerializer.Serialize(tours));
+            writer.Flush();
+            writer.Close();
```

Comment:
Dit laten lopen via repository. Als je in de tourservice dat doet met een write komt dat goed.
Regel 115(
        private void WriteTourData()
        {
            _repository.Write(_tours);
        })